Message/MessageFactory.php

```diff
@@ -88,6 +87,30 @@ protected function getDefaultOptions()
     }
 
     /**
+     * Find an email from database
+     *
+     * @param  string $reference
+     * @param  bool   $throwException
+     *
+     * @throws ReferenceNotFoundException
+     * @return  EmailInterface|null
+     */
+    public function getEmail($reference, $throwException = false)
+    {
+        if (!isset($this->emails[$reference])) {
+            $this->emails[$reference] = $this->em->getRepository($this->options['email_class'])->findOneByReference($reference);
+        }
+
+        $email = $this->emails[$reference];
+
+        if (true === $throwException) {
+            throw new ReferenceNotFoundException($reference, sprintf('Reference "%s" does not exist for email.', $reference));
```

Comment:
@cedric-g  : Because it was a mistake, it's fixed now

Concerning the throw exception toggle : You already sent an email exception telling that the reference doesn't exist on a "standard" use case . But it was probably a mistake to use a toggle. Do you prefer a try catch + method always returning an exception if reference doesn't exist?